Sources/SlidingContainerViewController.swift

```diff
@@ -0,0 +1,400 @@
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+import UIKit
+
+public enum MovementType {
+
+    case push
+    case hideUpperPushLower
+
+}
+
+public protocol ContainedViewController {
+
+    func didMinimizeControllerWith(diff: CGFloat)
+
+    func didMaximizeControllerWith(diff: CGFloat)
+
+    func updateFrameFor(heightPercentaje yPct: CGFloat, absolute diff: CGFloat)
+
+    var viewController: UIViewController { get }
+
+}
+
+extension ContainedViewController where Self: UIViewController {
+
+    public var viewController: UIViewController {
+        return self
+    }
+
+}
+
+public protocol SlidingContainerPresenter: class {
+
+    var movementType: MovementType { get }
+
+    func lowerController(for slidingContainer: SlidingContainerViewController) -> ContainedViewController
+
+    func upperController(for slidingContainer: SlidingContainerViewController) -> ContainedViewController
+
+    func upperViewMin(for slidingContainer: SlidingContainerViewController) -> CGFloat
+
+    func lowerViewMin(for slidingContainer: SlidingContainerViewController) -> CGFloat
+
+    func upperExtraDraggableArea(for slidingContainer: SlidingContainerViewController) -> CGFloat
+
+    func lowerExtraDraggableArea(for slidingContainer: SlidingContainerViewController) -> CGFloat
+
+}
+
+public protocol SlidingContainerDelegate: class {
+
+    func slidingContainerDidBeingDrag(_ slidingContainer: SlidingContainerViewController)
+
+    func slidingContainerDidEndDrag(_ slidingContainer: SlidingContainerViewController)
+
+}
+
+open class SlidingContainerViewController: UIViewController, UIGestureRecognizerDelegate {
+
+    static open let velTreshHold = 4000.0
+
+    @IBOutlet open weak var dragView: UIView?
+
+    @IBOutlet open weak var navView: UIView?
+
+    public weak var presenter: SlidingContainerPresenter!
+
+    public weak var delegate: SlidingContainerDelegate!
+
+    public var upperController: ContainedViewController!
```

Comment:
I think this should be private or at least its setter should. Same concern with the non-private properties below